File: Competitive Programming/Night.py
n = int(input())
t = 0
ini = 3
sum = 3
r = n

while r >= 3:
    while True:
        if r - ini >= 0:
            t = t + 1
            r = r - ini
            ini = ini + sum
            sum = sum + 1
        else:
            ini = 3
            sum = 3
            break

print(t,  r)

File: Competitive Programming/Night.cpp
#include <iostream>
using namespace std;

typedef long long int ll;

int main() {
    ll n, t = 0, ini = 3, sum = 3, r;

    cin >> n;
    r = n;

    while (r >= 3) {
        while (true) {
            if (r - ini >= 0) {
                t++;
                r -= ini;
                ini += sum;
                sum++;
            } else {
                ini = 3;
                sum = 3;
                break;
            }
        }
    }

    cout << t << " " << r << endl;

    return 0;
}